src/toast/templates/fourier2d.py

```diff
@@ -124,6 +124,8 @@ def _initialize(self, new_data):
 
             # Build up detector list
             for d in ob.local_detectors:
+                if ob.local_detector_flags[d] & self.det_flag_mask:
```

Comment:
Could we not use `ob.select_local_detectors()` here?

Reply:
I think I have now caught all of these and fixed them.